testnet/stacks-node/src/nakamoto_node/sign_coordinator.rs

```diff
@@ -758,8 +814,41 @@ impl SignCoordinator {
                         response_hash,
                         signature,
                     ))) => (response_hash, signature),
-                    SignerMessageV0::BlockResponse(BlockResponse::Rejected(_)) => {
-                        debug!("Received rejected block response. Ignoring.");
+                    SignerMessageV0::BlockResponse(BlockResponse::Rejected(rejected_data)) => {
+                        let Some(signer_entry) = &self.signer_entries.get(&slot_id) else {
+                            return Err(NakamotoNodeError::SignerSignatureError(
+                                "Signer entry not found".into(),
+                            ));
+                        };
+                        if rejected_data.signer_signature_hash
+                            == block.header.signer_signature_hash()
+                        {
+                            debug!(
+                                "Signer {} rejected our block {}/{}",
+                                slot_id,
+                                &block.header.consensus_hash,
+                                &block.header.block_hash()
+                            );
+                            total_reject_weight = total_reject_weight
+                                .checked_add(signer_entry.weight)
+                                .expect("FATAL: total weight rejected exceeds u32::MAX");
+
+                            if total_reject_weight.saturating_add(self.weight_threshold)
+                                > self.total_weight
+                            {
+                                debug!(
+                                    "{}/{} signers vote to reject our block {}/{}",
+                                    total_reject_weight,
+                                    self.total_weight,
+                                    &block.header.consensus_hash,
+                                    &block.header.block_hash()
+                                );
+                                counters.bump_naka_rejected_blocks();
+                                return Err(NakamotoNodeError::SignersRejected);
+                            }
+                        } else {
+                            debug!("Received rejected block response for a block besides my own. Ignoring.");
```

Comment:
Very nit but I prefer an early return/continue, just because it's a bit easier to read:

```rust
if rejected_data.signer_signature_hash != block.header.signer_signature_hash() {
    debug!("Received rejected block response for a block besides my own. Ignoring.");
    continue;
}
```